Default Assemblers

Sherpard opened the pull request:
This PR has a few targets ~~and it is still a Work In Progress (Not to be merged yet :) )~~

The flowing features are included:

- Default Assembler methods implemented on TupleBaseAssembler and BaseAssembler
Includes some documentation
- String qualifier at FluentAssembler DSL so it uses a Named annotation with the value specified

~~Fixes a bug at InMemorySequenceGeneratorIT~~

adrienlauer reviewed (approved):
Very nice PR. Overall, it simplifies practical use of assemblers. Thanks !

Sherpard committed:
FEAT: Add a default implementation to assemblers, so 1Way assemblers are a thing

FEAT: Adds String argument to assembler selector
FEAT: Add default assembler implementation

FIX: InMemorySequenceIT
	Test failed when executed on the whole suite, due to its Singleton nature.

Sherpard commented:
Cleaned up the code so it doesn't touch files that it shouldn't

adrienlauer commented:
Kind of fixes #69 because I don't think a real separation of the assembler interface is useful and it would complicate the assembler lookup code. This PR is more than enough.
